include/nil/blueprint/assigner.hpp

```diff
@@ -1727,12 +1668,17 @@ namespace nil {
             }
 
             template<typename InputType>
-            var put_into_assignment(InputType input, bool is_shared = false) { // TODO: column index is hardcoded but shouldn't be in the future
-                if (is_shared && maxNumProvers > 1) {
-                    return detail::put_shared<InputType, BlueprintFieldType, ArithmetizationParams, var>(input, assignments[currProverIdx]);
-                } else {
-                    return detail::put_constant<InputType, BlueprintFieldType, ArithmetizationParams, var>(input, assignments[currProverIdx]);
-                }
+            var put_constant_into_assignment(InputType input) {
+                return detail::put_constant<InputType, BlueprintFieldType, ArithmetizationParams, var>(input, assignments[currProverIdx]);
+            }
+
+            template<typename InputType>
+            var save_value(InputType input) {
```

Comment:
It would be nice to add docs to these functions. It's a bit hard to get what does function `save_value` does in the current architecture.